### ISSUE TITLE
Dots "." in commandLineArguments YAML are filtered out

### ISSUE THREAD

**Issue by eskerber** [yonaskolb/XcodeGen#321]
```yaml
commandLineArguments:
  "-this-works": false
  "-this.doesnt YES" : false
  "-thisWill Work Again": true
```

The resulting command line args I see when viewing the scheme are:

- [ ] this-works
- [x] thisWill Work Again

I tried to reproduce this in the tests, but I couldn't find (quickly) that allowed me to reproduce this from the YAML itself. Can work on a fix myself, if nudged in the right direction.

**Comment by yonaskolb** [comment]
From looking at the yaml posted above the fact that the offending line isn't green may give a hint. If you remove the space before the colon it will probably work
```diff
-"-this.doesnt YES" : false
+"-this.doesnt YES": false
```

Let me know if that works

**Comment by eskerber** [comment]
Apologies for the typo, that's not going to be it (in my source, I have about 7 of these values with properly placed colons).

Currently debugging, my dot-laden args args not surviving `Target.swiftL211`. Currently looking into what nefarious activity may be going on around `invalidItemBehaviour` 🤠 .

Thanks for the quick reply!

**Comment by yonaskolb** [comment]
Oh, without looking at it properly now I think it may be an issue with `JSONUtilities` in that it's treating dots as a dictionary lookup

**Comment by eskerber** [comment]
Yip. That'd be it!

![2018-06-21 at 11 09 pm](https://user-images.githubusercontent.com/75236/41757511-1be05ce0-75a8-11e8-8b32-bd13e127c29f.jpg)

Sleep will tell me if I should just have  our team #StopDoingThis. Feel like a fix won't exactly be straightforward. 

Maybe at least the docs can call it out.

**Comment by yonaskolb** [comment]
Having dots in keys should be supported. If it really is `JSONUtilities` the fork that this uses actually has a fix for that problem by specifying a `.key` instead of `.keyPath` (which it's use of `ExpressibleByStringLiteral` defaults to)
Feel free to submit a fix and test, if that's the issue

**Comment by yonaskolb** [comment]
It would be nice to support dots in all keys though. There's still an open issue about moving to `Codable` instead of `JSONUtilities` but it hasn't been done yet as it's not super easy because of the`[String: Any]` types and the merges and replacements. It should be possible though using something like https://github.com/yonaskolb/Codability and doing the data merging and massaging before decoding

**Comment by eskerber** [comment]
👍 great! Thanks for the insight. 

I couldn't quite grok all the diffs between your fork and the original, but XcodeGen is pointing to the latest commit of your fork at the moment. Are you saying the JSONUtilities fork should work, needs a minor update, or needs some cherry-pick or merge love from the original repo?

**Comment by yonaskolb** [comment]
I mean that the fork has support for looking up keys without being a path. That's not that scalable though as we'd have to update every call site in XcodeGen. Maybe the fork can be patched to change the default that `ExpressibleByStringLiteral` uses from `.keyPath` to `.key`. (or even an ugly static property somewhere that changes behaviour 😱)

**Comment by yonaskolb** [comment]
Fixed in #323

**Comment by eskerber** [comment]
Thank you thank you 🙏. Wish I could have jumped on this, you know your JSONUtils fork much better than I do obviously 😸
